notebook2/notebook/notebook.py

```diff
@@ -170,61 +198,201 @@ def external_url_for(path):
     return url + path
 
 
+# FIXME: use hash
+def UNSAFE_user_id_transform(user_id): return user_id.replace('|', '--_--')
+
+
+def read_svc_status(svc_name: str):
+    try:
+        # TODO: inspect exception for non-404
+        _ = k8s.read_namespaced_service(svc_name, 'default')
+        return 'Running'
+    except Exception:
+        return 'Deleted'
+
+
+def read_containers_status(container_statuses):
```

Comment:
I propose improvements to this be a separate PR. This PR is to get the MVP up, there are surely nicer interfaces to develop.